### DISCUSSION TITLE
Android Examples

### DISCUSSION THREAD

**Discussion by louwers** [maplibre/maplibre-native#1152] [General]
This discussion is a bit of a brainstorm how we can create examples that stand the test of time.

I am afraid that if we include a bunch of source code in Markdown files that is neither built nor tested, it will get out of date quickly.

1. There should be CI that builds and tests the example app.
1. Source code should be referenced instead of copy pasted into Markdown files. [Example](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-a-permanent-link-to-a-code-snippet)
1. Dependabot should be set up to automatically create PRs to update dependencies.
1. There should be instructions how to add a new example.
1. To be decided: write documentation in in-source Markdown files or use a documentation generator and deploy.

**Comment by louwers** [comment]
Since we already have [MapboxGLAndroidSDKTestApp](https://github.com/maplibre/maplibre-native/tree/main/platform/android/MapboxGLAndroidSDKTestApp) which is very well tested, it may make sense to use the examples from this app.

A great benefit here is that this app is built and tested on CI.

A downside is that some features of these app are really more for developers of the SDK than for end-users.

We also probably don't want to say "here go check out this source code" since often an example requires more than just the source code. E.g. pulling some data from somewhere. Writing explanations about concepts and APIs involved. Etc.

**Comment by birkskyum** [comment]
Do we include source code in markdown files now? And if so - where?

**Reply by louwers** [reply]
We started to do this for the Android examples. https://github.com/maplibre/maplibre-native/blob/main/docs/mdbook/src/android/getting-started-guide.md

Maybe we can use literate programming instead: https://github.com/jillesvangurp/kotlin4example

**Comment by louwers** [comment]
Issue from @fynngodau who would like to work on this: https://github.com/maplibre/maplibre-plugins-android/issues/27
